src/highdicom/map/sop.py

```diff
@@ -0,0 +1,767 @@
+
+            # Real World Value Mapping
+            if real_world_value_mappings:
+                rwvm = real_world_value_mappings[i]
+                # If the input RWVM is a list of individual mappings we need to
+                # assign those to a sequence as we go. Otherwise, if it is a
+                # list of sequences of RWVMs, then we can just assign them
+                # outright
+                if isinstance(rwvm, RealWorldValueMapping):
+                    pffg_item.RealWorldValueMappingSequence = [rwvm]
+                else:
+                    pffg_item.RealWorldValueMappingSequence = rwvm
+
+                self.PerFrameFunctionalGroupsSequence.append(pffg_item)
+
+            self._append_pixel_data(pixel_array[i, ...])
+
+    def _append_pixel_data(self, plane: np.ndarray) -> None:
+        """Appends the provided array of pixels to the pixel data element.
+
+        Depending on the data type, the pixel data may be stored in either the
+        Pixel Data, Float Pixel Data, or Double Float Pixel Data element.
+
+        Parameters
+        ----------
+        plane: np.ndarray
+            Two dimensional array of pixels of an individual frame (plane)
+
+        Raises
+        ------
+        ValueError
+            When the input pixel array's dtype does not match that of the
+            SOP Instance
+
+        """
+        pixel_data_type, pixel_data_attr = self._get_pixel_data_type_and_attr(
+            plane
+        )
+        if self._pixel_data_type != pixel_data_type:
+            raise ValueError(
+                'Data type of input pixel array '
+                'does not match that of SOP instance. '
+                f'Expected "{self._pixel_data_type}", got "{pixel_data_type}".'
+            )
+
+        pixel_data_bytes = getattr(self, pixel_data_attr)
+
+        # Before adding new pixel data, remove trailing null padding byte
+        if len(pixel_data_bytes) == get_expected_length(self) + 1:
+            pixel_data_bytes = pixel_data_bytes[:-1]
+
+        # Add new pixel data
+        if self.file_meta.TransferSyntaxUID.is_encapsulated:
+            # To add a new frame item to the encapsulated Pixel Data element
+            # we first need to unpack the existing frames, then add the pixel
+            # data of the current plane, and finally encapsulate all frames.
+            all_frames = decode_data_sequence(pixel_data_bytes)
+            all_frames.append(self._encode_pixels(plane))
+            pixel_data_bytes = encapsulate(all_frames)
+        else:
+            pixel_data_bytes += self._encode_pixels(plane)
+
+        # FIXME(cg): This fails with the latest release of PyDICOM because the
+        # when using Double Float Pixel Data. The bug is fixed in master but
+        # has not been released yet.
+        # See https://github.com/pydicom/pydicom/pull/1413 for more details
+        setattr(self, pixel_data_attr, pixel_data_bytes)
```

Comment:
Just so we can track it in the review, this fails on the latest release of PyDICOM when using Double Float Pixel Data. The bug is fixed in their master branch, though.

https://github.com/pydicom/pydicom/pull/1413

Reply:
Resolving this comment, since the issue will be fixed with the upcoming 2.2 release.